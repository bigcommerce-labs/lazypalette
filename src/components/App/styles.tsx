import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
