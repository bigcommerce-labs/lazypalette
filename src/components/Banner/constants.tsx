import React from 'react';

export const CopyMessage = 'Link copied to clipboard.';

export const TooltipMessage =
    <>
        <p>You must launch your store from the Getting Started page in order to make it public.</p>
        <p><a href="/manage/getting-started">Go to Getting Started</a></p>
        <p>You can still let other people see your store before it is launched by sharing the private store link.</p>
    </>;

export const MaintenanceMessage = 'Your store is down for maintenance.';
export const PrelaunchMessage = 'Your store has not launched and is not publicly visible.';
