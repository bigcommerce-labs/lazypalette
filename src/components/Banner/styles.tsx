import styled from 'styled-components';

export const StyledBanner = styled.div`
    background-color: #FFD24D;
    text-align: center;
    width: 100%;
    line-height: 2.5rem;
    font-size: 1rem;
    font-weight: 600;
    color: #303540;
`;

export const StyledIcon = styled.span`
    span {
        margin-bottom: -5px;
    }
`;

export const StyledMessage = styled.p`
    text-align: left;

    a {
        color: ${({ theme }) => theme.colors.primary};
        display: block;
        font-weight: bold;
        margin-top: 5px;
        text-decoration: none;
    }
`;

export const StyledShareableLink = styled.button`
    background: ${({ theme }) => theme.colors.warning};
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold}
    font-family: ${({ theme }) => theme.typography.fontFamily.headings};
    margin: .5rem 0 .5rem 1.25rem;
    outline: none;
    border: 1px solid;
    border-radius: 2px;
    display: inline-block;
    padding: .25rem .5rem;
    line-height: 1.375rem;
    cursor: pointer;
`;
