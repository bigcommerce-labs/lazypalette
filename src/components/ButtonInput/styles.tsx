import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0;
`;

interface ButtonProps {
    border?: boolean;
    classType?: string;
}

export const Button = styled.button.attrs<ButtonProps>({})`
    height: 2rem;
    border-radius: 2px;
    box-sizing: border-box;
    border: ${({ border, theme }) => border ? `1px solid ${theme.colors.primary}` : 'none'};
    padding: ${({ border }) => border ? '0 1rem' : '0'};
    font-family: ${({ theme }) => theme.typography.fontFamily.sans };
    font-size: 1rem;
    cursor: pointer;

    ${({ border, classType, theme }) => {
        const isPrimary = classType === 'primary';

        return `
            background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
            color: ${isPrimary ? '#FFF' : theme.colors.primary};

            :hover {
                background-color: ${isPrimary ? theme.colors.primaryHover : theme.colors.secondaryHover};
                ${!border && `
                    background-color: transparent;
                `}
                border-color: ${isPrimary ? theme.colors.brandPrimary : theme.colors.primary};
                text-decoration: ${border ? 'none' : 'underline'};
            }

            :focus {
                outline: none !important;
            }

            :disabled {
                background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
                opacity: 0.5;
            }
        `;
    }}
`;

Button.defaultProps = {
    theme: {
        colors: {
            background: '#F5F7FA',
            brandPrimary: '#273A8A',
            primary: '#3F69FE',
        },
        typography: {
            fontFamily: {
                sans: '"Source Sans Pro", "Helvetica Neue", Arial, sans-serif',
            },
        },
    },
};
