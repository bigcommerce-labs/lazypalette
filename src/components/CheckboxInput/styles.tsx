import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
`;

export const Input = styled.input.attrs({
    type: 'checkbox',
})`
    ${({ theme }) => `
        opacity: 0;

        :not(:checked) + label:after {
            opacity: 0;
        }

        :not(:checked) + label {
            background: #FFF;
            border: 1px solid ${theme.colors.guideText};
            box-sizing: border-box;
        }
    `}
`;

export const Label = styled.label`
    ${({ theme }) => `
        cursor: pointer;
        position: absolute;
        width: 1.25rem;
        height: 1.25rem;
        top: 25%;
        left: 0;
        background: ${theme.colors.primary};
        border-radius: 2px;

        :after {
            content: '';
            position: absolute;
            width: .5rem;
            height: .25rem;
            background: transparent;
            top: .25rem;
            left: .25rem;
            border: 3px solid ${theme.colors.stroke};
            border-top: none;
            border-right: none;
            transform: rotate(-45deg);
        }
    `}
`;

export const HiddenLabel = styled.label`
    padding-left: .5rem;
    line-height: 1.25rem;
    display: inline-block;
`;
