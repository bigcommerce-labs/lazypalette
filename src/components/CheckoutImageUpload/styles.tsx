import styled from 'styled-components';

export const UploadButtonLabel = styled.label`
    height: 2rem;
    border-radius: 2px;
    border: 1px solid ${({ theme }) => theme.colors.primary};
    padding: 0 0.75em;
    font-size: 0.75rem;
    cursor: pointer;
    display: inline-block;
    color: ${({ theme }) => theme.colors.primary};
    margin-right: 1rem;
`;

export const CheckoutImageUploadWrapper = styled.div`
    margin: 1rem 0;
`;

export const CheckoutImageInputWrapper = styled.div`
    margin: 1rem 0;
`;

export const CheckoutImagePreviewWrapper = styled.div``;

export const CheckoutImageUploadError = styled.p`
    color: ${({ theme }) => theme.colors.error};
`;

export const CheckoutImageRemoveLink = styled.a`
    color: ${({ theme }) => theme.colors.error};
`;
