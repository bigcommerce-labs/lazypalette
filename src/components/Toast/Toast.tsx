import { theme, Icon } from 'pattern-lab';
import React, { MouseEvent, MouseEventHandler, PureComponent } from 'react';

import { CloseIcon, Container, Message, ToastBox } from './styles';

interface ToastProps extends Partial<{
    autoDismiss: boolean;
    type: string;
    children: JSX.Element | string;
    onClose: MouseEventHandler<HTMLDivElement>;
}> {}

class Toast extends PureComponent<ToastProps> {
    static defaultProps: ToastProps = {
        autoDismiss: true,
    };

    handleClose = (event: MouseEvent<HTMLDivElement>) => {
        if (this.props.onClose) {
            this.props.onClose(event);
        }
    };

    render() {
        const { autoDismiss, type, children } = this.props;

        return (
            <Container autoDismiss={autoDismiss}>
                <ToastBox type={type}>
                    <Message>{children}</Message>
                    <CloseIcon onClick={this.handleClose}>
                        <Icon
                            glyph="closeX"
                            primaryColor={theme.colors.background}
                            size="small"
                        />
                    </CloseIcon>
                </ToastBox>
            </Container>
        );
    }
}

export default Toast;
