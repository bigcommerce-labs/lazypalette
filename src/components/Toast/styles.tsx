import styled, { keyframes } from 'styled-components';

import { ToastTimeout, ToastType } from '../../actions/constants';

interface ContainerProps {
    autoDismiss?: boolean;
}

export const delayOut = keyframes`
    from { opacity: 1; }
    to { opacity: 0; top: -4rem; }
`;

const delayOutDuration = 250;

export const Container = styled.div.attrs<ContainerProps>({})`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%
    display: flex;
    opacity: 1;
    justify-content: center;
    animation: ${delayOut} ${delayOutDuration}ms linear;
    animation-delay: calc(${ToastTimeout.Duration}ms - ${delayOutDuration}ms);
    animation-fill-mode: forwards;

    ${({ autoDismiss }) => autoDismiss === false && `
        animation: none;
    `};
`;

export const slideIn = keyframes`
    from { top: -4rem; }
    to { top: .5rem; }
`;

interface ToastBoxProps {
    type?: string;
}

export const ToastBox = styled.div.attrs<ToastBoxProps>({})`
    position: relative;
    top: .5rem;
    display: flex;
    background-color: ${({ theme }) => theme.colors.primaryText};
    color: ${({ theme }) => theme.colors.empty};
    font-weight: ${({ theme }) => theme.typography.fontWeight.normal};
    font-size: .875rem;
    padding: .75rem .25rem .75rem .75rem;
    border-radius: 2px;
    box-shadow: ${({ theme }) => theme.elevation.raised};
    animation: ${slideIn} 250ms linear;

    ${({ theme, type }) => type === ToastType.Invalid && `
        background-color: ${theme.colors.error};
    `}
`;

export const CloseIcon = styled.div`
    text-decoration: none;
    cursor: pointer;
    margin-left: 1.5rem;

    :after {
        content: 'âœ•';
        color: ${({ theme }) => theme.colors.empty};
        padding: .25rem .5rem;

        :visited, :hover, :active {
            color: ${({ theme }) => theme.colors.primaryText};
        }
    }
`;
