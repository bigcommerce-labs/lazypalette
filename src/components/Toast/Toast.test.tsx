import { shallow } from 'enzyme';
import React from 'react';

import { ToastMessages, ToastType } from '../../actions/constants';

import { CloseIcon } from './styles';
import Toast from './Toast';

describe('Toast', () => {
    const mockOnClose = jest.fn();

    it('renders', () => {
        const toast = shallow(
            <Toast
                autoDismiss
                children={ToastMessages.SaveTheme}
                onClose={mockOnClose}
                type={ToastType.Success}
            />
        );

        expect(toast).toMatchSnapshot();
    });

    describe('close toast button', () => {
        const toast = shallow(
            <Toast
                autoDismiss
                children={ToastMessages.SaveTheme}
                onClose={mockOnClose}
                type={ToastType.Success}
            />
        );

        it('should call onClose prop when clicked', () => {
            const closeButton = toast.find(CloseIcon);

            closeButton.simulate('click');
            expect(mockOnClose).toHaveBeenCalledTimes(1);
        });
    });
});
