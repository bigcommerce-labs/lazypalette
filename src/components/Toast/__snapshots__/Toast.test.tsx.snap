// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders 1`] = `
<styled.div
  autoDismiss={true}
>
  <styled.div
    type="Success"
  >
    <styled.span>
      Changes have been saved.
    </styled.span>
    <styled.div
      onClick={[Function]}
    >
      <Icon
        glyph="closeX"
        primaryColor="#F5F7FA"
        size="small"
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
